runtime/src/hardened_unpack.rs

```diff
@@ -170,8 +172,15 @@ where
         }
         let target = target.unwrap();
 
-        let unpack = entry.unpack(target);
-        check_unpack_result(unpack.map(|_unpack| true)?, path_str)?;
+        if re_appendvec_filename.is_match(&*path_str) && target.exists() {
+            info!(
+                "The accounts target {} exists.  Skip unpacking",
+                target.display()
+            );
+        } else {
+            let unpack = entry.unpack(target);
+            check_unpack_result(unpack.map(|_unpack| true)?, path_str)?;
+        }
```

Comment:
How much time does this save? In the issue, I assumed we'd either unpack all appendvecs or none. Verify that we have them all from the accounts fields we unpack from the snapshot, but we can stop unpacking immediately once we get those accounts fields.

Reply:
we'd either unpack all appendvecs or none because bank.verify_snapshot_hash is called at the bank level.